core/src/main/java/com/netease/arctic/optimizing/OptimizingDataReader.java

```diff
@@ -21,10 +21,23 @@
 import org.apache.iceberg.data.Record;
 import org.apache.iceberg.io.CloseableIterable;
 
+/**
+ * An interface to read the data and delete data.
+ */
 public interface OptimizingDataReader {
+
+  /**
+   * Reading data of optimizing task, If a task contains a delete operation, then the read result is MOR.
+   */
   CloseableIterable<Record> readData();
 
+  /**
+   * Reading data that needs to be deleted during MOR. If there is no delete file, then the result is empty.
+   */
   CloseableIterable<Record> readDeletedData();
 
+  /**
+   * Close the reader
+   */
```

Comment:
Would it be better to extend `java.io.Closeable`?

Reply:
yeah